// @metamask/auto-changelog/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateChangelog = exports.ChangelogFormattingError = exports.updateChangelog = exports.parseChangelog = exports.createEmptyChangelog = exports.Changelog = void 0;
var changelog_1 = require("./changelog");
Object.defineProperty(exports, "Changelog", { enumerable: true, get: function () { return __importDefault(changelog_1).default; } });
var init_1 = require("./init");
Object.defineProperty(exports, "createEmptyChangelog", { enumerable: true, get: function () { return init_1.createEmptyChangelog; } });
var parse_changelog_1 = require("./parse-changelog");
Object.defineProperty(exports, "parseChangelog", { enumerable: true, get: function () { return parse_changelog_1.parseChangelog; } });
var update_changelog_1 = require("./update-changelog");
Object.defineProperty(exports, "updateChangelog", { enumerable: true, get: function () { return update_changelog_1.updateChangelog; } });
var validate_changelog_1 = require("./validate-changelog");
Object.defineProperty(exports, "ChangelogFormattingError", { enumerable: true, get: function () { return validate_changelog_1.ChangelogFormattingError; } });
Object.defineProperty(exports, "validateChangelog", { enumerable: true, get: function () { return validate_changelog_1.validateChangelog; } });
