// @moralisweb3/common-streams-utils/lib/dataTypes/StreamTrigger/StreamTrigger.mock.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.mockStreamTrigger = void 0;
var balanceOfAbi = {
    constant: true,
    inputs: [
        {
            name: 'owner',
            type: 'address',
        },
    ],
    name: 'balanceOf',
    outputs: [
        {
            name: 'toBalance',
            type: 'uint256',
        },
    ],
    payable: false,
    stateMutability: 'view',
    type: 'function',
};
var allowanceAbi = {
    constant: true,
    inputs: [
        {
            name: 'owner',
            type: 'address',
        },
        {
            name: 'spender',
            type: 'address',
        },
    ],
    name: 'balanceOf',
    outputs: [
        {
            name: 'remaining',
            type: 'uint256',
        },
    ],
    payable: false,
    stateMutability: 'view',
    type: 'function',
};
var erc20TransferTrigger = {
    type: 'erc20transfer',
    contractAddress: '0x95aD61b0a150d79219dCF64E1E6Cc01f0B64C4cE',
    functionAbi: balanceOfAbi,
    inputs: ['$to'],
};
var allowanceTrigger = {
    type: 'erc20transfer',
    contractAddress: '$contract',
    functionAbi: allowanceAbi,
    inputs: [['$from', '$to']],
};
exports.mockStreamTrigger = {
    ERC20_TRANSFER: erc20TransferTrigger,
    ALLOWANCE_SELECTORS: allowanceTrigger,
};
